django/core/files/storage.py

```diff
@@ -248,6 +250,14 @@ def _value_or_setting(self, value, setting):
 
     @cached_property
     def base_location(self):
+        # Check if django test is running and if test media root is present.
+        if sys.argv[1] == "test":
+            if hasattr(settings, "TEST_MEDIA_ROOT"):
+                return settings.TEST_MEDIA_ROOT
+            else:
+                warnings.warn("TEST_MEDIA_ROOT not set, MEDIA_ROOT will be used. Test media will not be deleted after the test run.")
```

Comment:
In addition to the comment on the ticket -- there should be a very good reason for creating an experience where, when you do something the first time with the defaults, you get a warning. Backwards compatibility with unwanted behavior is, IMO, not a very good reason.